### PR TITLE
EVG-2470 Add structured logging to ec2 host provider

### PR DESCRIPTION
**Pull request by bsamek**

### CODE REVIEW

File: cloud/new_ec2.go [hunk 1]
```diff
@@ -128,6 +128,12 @@ func (m *ec2Manager) spawnOnDemandHost(h *host.Host, ec2Settings *NewEC2Provider
...
 
+	grip.Debug(message.Fields{
+		"msg":     "starting on-demand instance",
+		"args":    input,
+		"host_id": h.Id,
+		"distro":  h.Distro.Id,
+	})
 	reservation, err := m.client.RunInstances(input)
 	if err != nil || reservation == nil {
 		grip.Error(errors.Wrapf(h.Remove(), "error removing intent host %s", h.Id))
```
Comment (tychoish) [inline cloud/new_ec2.go]: can you add message.WrapError here too?

File: cloud/new_ec2.go [hunk 2]
```diff
@@ -128,6 +128,12 @@ func (m *ec2Manager) spawnOnDemandHost(h *host.Host, ec2Settings *NewEC2Provider
 		input.SecurityGroups = []*string{&ec2Settings.SecurityGroup}
 	}
 
+	grip.Debug(message.Fields{
```
Comment (tychoish) [inline cloud/new_ec2.go]: should we also add host provider name? (throughout?)

File: cloud/new_ec2.go [hunk 3]
```diff
@@ -338,6 +376,11 @@ func (m *ec2Manager) TerminateInstance(h *host.Host) error {
 	if m.spot(h) {
 		canTerminate, err := m.cancelSpotRequest(h)
 		if err != nil {
+			grip.Error(message.WrapError(err, message.Fields{
+				"msg":     "error terminating spot instance",
+				"host_id": h.Id,
+				"distro":  h.Distro.Id,
+			}))
 			return errors.Wrap(err, "error canceling spot request")
```
Comment (tychoish) [inline cloud/new_ec2.go]: should we wrap the error before logging?

File: cloud/new_ec2.go [hunk 4]
```diff
@@ -373,21 +423,32 @@ func (m *ec2Manager) cancelSpotRequest(h *host.Host) (bool, error) {
...
+			"msg":     "error getting spot request info",
+			"host_id": h.Id,
+			"distro":  h.Distro.Id,
+		}))
+		return false, errors.Wrapf(err, "failed to get spot request info for %s", h.Id)
 	}
 	grip.Info(message.Fields{
 		"msg":     "canceling spot request",
-		"spot_id": h.Id,
+		"host_id": h.Id,
```
Comment (tychoish) [inline cloud/new_ec2.go]: I'd probably change host_id to host

File: cloud/new_ec2.go [hunk 5]
```diff
@@ -373,21 +423,32 @@ func (m *ec2Manager) cancelSpotRequest(h *host.Host) (bool, error) {
...
+		"distro":  h.Distro.Id,
 	})
 	if _, err = m.client.CancelSpotInstanceRequests(&ec2.CancelSpotInstanceRequestsInput{
 		SpotInstanceRequestIds: []*string{makeStringPtr(h.Id)},
 	}); err != nil {
+		grip.Error(message.Fields{
+			"msg":     "failed to cancel spot request",
```
Comment (tychoish) [inline cloud/new_ec2.go]: change "msg" to "message" everywhere.

File: cloud/new_ec2.go (hunk 2 above)
Comment (bsamek) [reply cloud/new_ec2.go]: done

File: cloud/new_ec2.go (hunk 1 above)
Comment (bsamek) [reply cloud/new_ec2.go]: done

File: cloud/new_ec2.go (hunk 3 above)
Comment (bsamek) [reply cloud/new_ec2.go]: I'm not sure what you mean. Don't use `message.WrapError`?

File: cloud/new_ec2.go (hunk 4 above)
Comment (bsamek) [reply cloud/new_ec2.go]: done

File: cloud/new_ec2.go (hunk 5 above)
Comment (bsamek) [reply cloud/new_ec2.go]: done

**Pull request merged**